lib/rex/compat.rb

```diff
@@ -198,7 +198,7 @@ def self.open_webrtc_browser(url='http://google.com/')
   else
     # If DISPLAY is not set, most unix graphical tools will not work.
     if ENV['PATH'] && ENV['DISPLAY']
-      ['google-chrome', 'chrome', 'chromium', 'firefox' , 'firefox', 'opera'].each do |browser|
+      [ENV['MSF_BROWSER_BIN'], 'google-chrome', 'chrome', 'chromium', 'firefox', 'opera'].compact.each do |browser|
```

Comment:
What if we set `xdg-open` after `MSF_BROWSER_BIN`? So, that in case an user is not aware of the environment variable, then the file will still open in a browser.

Reply:
> My concern with this implementation right now is that this isn't documented anywhere, so you are relying on people knowing that the environment variable MSF_BROWSER_BIN exists and then also setting it.

I believe the plan was to add a note about this to the framework docs once the gem is being updated in a framework PR

> I also wonder if the environment variable is not a string, then will this cause issues?

This isn't an issue :+1:

`nil` is handled via the `compact`, and it's not possible to set non-string values in the ENV object in Ruby

```
2.7.2 :001 > ENV['a'] = 3
(irb):1:in `[]=': no implicit conversion of Integer into String (TypeError)
```

Reply:
@JitPatro I believe you're the maintainer of the Metasploit snap package? Is it possible to specify environment variables such as `MSF_BROWSER_BIN` for this scenario?

Presumably here? https://github.com/JitPatro/metasploit-framework-snap/blob/161ce782a6e73703c35683ea43092b0165ee9f67/snap/snapcraft.yaml#L32-L42

Or if there's a more conventional way to solve this problem with snap packages, that would be interesting to hear about too :+1:

Reply:
Thanks to you @adfoster-r7 I found the best way to solve this problem. We can use `xdg-settings` to set a default setting for opening all file types in a browser with this command:-
```
xdg-settings set default-url-scheme-handler file browser_name.desktop
```
We can get the user's default browser with this command:-
```
xdg-settings get default-web-browser
```

I cannot comment on how you guys should deal with this problem as I don't work for rapid7, but I'm going to modify the Metasploit snap package to open all files in a browser. 🙂 





> Or if there's a more conventional way to solve this problem with snap packages, that would be interesting to hear about too +1

Yes, there is really a nice option available to Snap developers to achieve this. I can add a configuration option which can be set by users at runtime, for e.g, `snap set MSF_BROWSER_BIN /path/to/browser/binary`